### DISCUSSION TITLE
Is there a way to configure polymorphic_union on ConcreteBase

### DISCUSSION THREAD

**Discussion by lovetox** [sqlalchemy/sqlalchemy#7255] [Usage Questions] [answered]
polymorphic_union() has an arg `casts_null` which is essential on Oracle Databases as CLOB fields can not be cast to NULL.

It seems its not possible when i use ConcreteBase to influence this setting.

Further casts_null is defined as legacy setting, and was introduced in 0.7,
in light of upcoming 2.0 is this not maybe a possibility to not cast to NULL anymore?

**Comment by zzzeek** [comment, marked as answer]
from my reading of the string, it states that "casts_null", which defaults to True, needs to be set to False on Oracle, meaning, we dont render the casts.  IOW, the fact that "casts null" happens at all is the legacy setting, so if we did anything in 2.0 , we'd take out the whole "CAST X AS NULL" thing, which IIUEC means Oracle would work without having to set anything.

you can override how polymorphic_union is called by ConcreteBase by overriding the "_create_polymorphic_union" method of the class.  it's not documented right now but nothing is changing there without us formally adding a supported feature or something like that:

```py
    @classmethod
    def _create_polymorphic_union(cls, mappers, discriminator_name):
        return polymorphic_union(
            OrderedDict(
                (mp.polymorphic_identity, mp.local_table) for mp in mappers
            ),
            discriminator_name,
            "pjoin",
        )
```

**Reply by zzzeek** [reply]
yeah the docstring says, "this is a legacy behavior", meaning the behavior of casting to null is what's legacy.  if it said, "this is a legacy **parameter** or **setting**", that would indicate the parameter going away which is what you don't want.

this is a seldom used API feature so we didnt have anything slated to change in 2.0 for this, though maybe we want to get rid of the "casts null" part.

**Reply by lovetox** [reply]
thanks for the hint with overriding, i didnt even consider that because its very ugly and prone to breakage,
though short term i think this works for me.

In my opinion there are 2 possible ways forward long term

1. remove the arg and not cast anymore
2. make it possible everywhere to pass that arg

Are you ok with me going forward and open an issue for that?

Thanks

**Reply by zzzeek** [reply]
I think the arg should default to False but stay around in case someone wants that cast thing. ConcreteBase then has "_concrete_discriminator_name" to set the name of the alias, that's one precedent, but i wouldnt want "_concrete_x", "_concrete_y" all down the class.  probably just renaming the method _concrete_create_polymorphic_union and documenting it that someone can override it in the rare case they'd want cast_nulls for now.
